File: app/app/src/main/java/me/williamhester/reddit/convert/ConverterException.java
package me.williamhester.reddit.convert;

import com.google.gson.JsonElement;

/** An exception thrown when we failed to parse the response. */
public class ConverterException extends Exception {
  public ConverterException(String message) {
    super(message);
  }

  public ConverterException(JsonElement object, String expectedType) {
    super(expectedType + ", got:\n" + object.toString());
  }
}

File: app/app/src/main/java/me/williamhester/reddit/convert/ConverterException.kt
package me.williamhester.reddit.convert

import com.google.gson.JsonElement

/** An exception thrown when we failed to parse the response.  */
class ConverterException : Exception {
  constructor(message: String) : super(message)
  constructor(element: JsonElement, expectedType: String) :
      super(expectedType + ", got:\n" + element.toString())
}
